In [1]:
import pandas as pd

# Load the dataset
df = pd.read_csv('/Users/maazhussain/Desktop/Projects/AB Testing & Experiment /ab_test_data.csv')

# Check basic conversion stats
conversion_rates = df.groupby('group')['converted'].agg(['mean', 'count', 'sum']).rename(columns={'mean': 'conversion_rate'})
conversion_rates


conversion_rate  count  sum
group                                 
control           0.126531    490   62
treatment         0.182353    510   93

In [3]:
from statsmodels.stats.proportion import proportions_ztest

# Successes and observations
success = [62, 93]              # Converted
n_obs = [490, 510]              # Total users

# One-tailed test (treatment > control)
z_stat, p_val = proportions_ztest(success, n_obs, alternative='larger')

print(f"Z-statistic: {z_stat:.4f}")
print(f"P-value: {p_val:.4f}")


Z-statistic: -2.4384
P-value: 0.9926


📊 A/B Testing – Conversion Experiment Report

🧭 Objective

To evaluate whether a new treatment (e.g., updated design, messaging, or offer) leads to a higher user conversion rate compared to the existing control group.

📁 Dataset Summary

Total Users: 1000 (490 control, 510 treatment)
Features:
user_id: Unique identifier
group: A/B group assignment (control, treatment)
converted: Conversion outcome (1 = converted, 0 = not)

🔬 Hypotheses

Null Hypothesis (H₀):
The conversion rate of the treatment group is equal to or less than the control group.
Alternative Hypothesis (H₁):
The treatment group has a higher conversion rate.

📈 Observed Conversion Rates

Group	Conversion Rate	Users	Conversions
Control	12.65%	490	62
Treatment	18.24%	510	93

🧪 Test Performed

Test Type: Two-sample proportions z-test (one-tailed)
Z-statistic: -2.4384
P-value: 0.9926

✅ Conclusion

The p-value (0.9926) is much greater than the 0.05 threshold. Therefore, we fail to reject the null hypothesis.
➡️ There is no statistically significant evidence that the treatment group performs better than the control group. The observed uplift is likely due to chance.